editor/src/core/shared/github.spec.ts

```diff
@@ -62,3 +79,372 @@ if (root != null) {
     `)
   })
 })
+
+describe('mergeProjectContentsTree', () => {
+  it('should always return a valid result', () => {
+    const parameters = FastCheck.tuple(
+      pathArbitrary(),
+      FastCheck.oneof(projectContentsTreeArbitrary(), FastCheck.constant(null)),
+      FastCheck.oneof(projectContentsTreeArbitrary(), FastCheck.constant(null)),
+      FastCheck.oneof(projectContentsTreeArbitrary(), FastCheck.constant(null)),
+    )
+    function checkMergeProjectContentsTree([
+      fullPath,
+      currentContents,
+      originContents,
+      branchContents,
+    ]: [
+      string,
+      ProjectContentsTree | null,
+      ProjectContentsTree | null,
+      ProjectContentsTree | null,
+    ]): boolean {
+      mergeProjectContentsTree(2000000, fullPath, currentContents, originContents, branchContents)
+      return true
+    }
+    const property = FastCheck.property(parameters, checkMergeProjectContentsTree)
+    FastCheck.assert(property, { verbose: true })
+  })
+  it('should return the current contents when a change only exists for that', () => {
+    const originContents = projectContentFile(
+      '/myfile.txt',
+      textFile(
+        textFileContents('origin code', unparsed, RevisionsState.CodeAhead),
+        null,
+        null,
+        1000000,
+      ),
```

Comment:
You can use `codeFile` to simplify this: <https://github.com/concrete-utopia/utopia/blob/e55cb7c6e431f0eb9c01b01709b3dbdd6cd7f397/editor/src/core/shared/project-file-types.ts#L574-L583>

Reply:
Fixed in fe952b182.